draft-ietf-taps-impl.md

```diff
@@ -540,6 +540,143 @@ It is also possible that protocol stacks within a particular leaf node use 0-RTT
 
 0-RTT handshakes often rely on previous state, such as TCP Fast Open cookies, previously established TLS tickets, or out-of-band distributed pre-shared keys (PSKs). Implementations should be aware of security concerns around using these tokens across multiple addresses or paths when racing. In the case of TLS, any given ticket or PSK should only be used on one leaf node. If implementations have multiple tickets available from a previous connection, each leaf node attempt must use a different ticket. In effect, each leaf node will send the same early application data, yet encoded (encrypted) differently on the wire.
 
+# Implementing Message Framers
+
+Message Framers are pieces of code that define simple transformations
+between application Message data and raw transport protocol data. A Framer
+can encapsulate or encode outbound Messages, and decapsulate or decode
+inbound data into Messages.
+
+While many protocols can be represented as Message Framers, for the
+purposes of the Transport Services interface these are ways for applications
+or application frameworks to define their own Message parsing to be
+included within a Connection's Protocol Stack. As an example, TLS can
+serve the purpose of framing data over TCP, but is exposed as a protocol
+natively supported by the Transport Services interface.
+
+Most Message Framers fall into one of two categories:
+
+- Header-prefixed record formats, such as a basic Type-Length-Value (TLV) structure
+
+- Delimiter-separated formats, such as HTTP/1.1.
+
+Common Message Framers can be provided by the Transport Services implementation,
+but an implemention ought to allow custom Message Framers to be defined by
+the application or some other piece of software. This section describes one
+possible interface for defining Message Framers as an example.
+
+## Defining Message Framers
+
+A Message Framer is primarily defined by the set of code that handles events
+for a framer implementation, specifically how it handles inbound and outbound data
+parsing. One way to represent a Message Framer is as an object that an application
+or other piece of software can interact with to receive events for parsing and sending
+data. Such Message Framer objects will deliver events to code that is written either as
+part of the application or a helper library. This piece of code that implements custom
+framing logic will be referred to as the "framer implementation".
```

Comment:
How does "Message Framer objects will deliver events to code that is written either as part of the application or a helper library" relate to the Transport Services implementation providing common framers as mentioned earlier?

Reply:
Good call. Reworded this section!